modules/pipeline/pipengine/reconciler/policy.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reconciler
+
+import (
+	"fmt"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/modules/pipeline/dbclient"
+	"github.com/erda-project/erda/modules/pipeline/spec"
+	"github.com/erda-project/erda/pkg/parser/pipelineyml"
+)
+
+type PolicyHandlerOptions struct {
+	dbClient *dbclient.Client
+}
+
+var policyHandlers = map[apistructs.PolicyName]func(*spec.PipelineTask, PolicyHandlerOptions) (*spec.PipelineTask, error){
+	apistructs.LastResultPolicyType:           handlerLastResultPolicyType,
+	apistructs.InPipelineLastResultPolicyType: handlerInPipelineLastResultPolicyType,
+	apistructs.LastSuccessResultPolicyType:    handlerLastSuccessResultPolicyType,
+	apistructs.FactoryNewPolicyType:           handlerFactoryNewPolicyType,
+}
+
+func handlerLastResultPolicyType(task *spec.PipelineTask, opt PolicyHandlerOptions) (result *spec.PipelineTask, err error) {
+	return task, nil
+}
+
+func handlerInPipelineLastResultPolicyType(task *spec.PipelineTask, opt PolicyHandlerOptions) (result *spec.PipelineTask, err error) {
+	return task, nil
+}
+
+func handlerLastSuccessResultPolicyType(task *spec.PipelineTask, opt PolicyHandlerOptions) (result *spec.PipelineTask, err error) {
+	if !task.IsSnippet {
+		return task, nil
+	}
+
+	if task.Extra.Action.SnippetConfig == nil {
+		return task, nil
+	}
+
+	ymlName, source := getPipelineSourceAndNameBySnippetConfig(task.Extra.Action.SnippetConfig)
+
+	runSuccessPipeline, _, _, _, err := opt.dbClient.PageListPipelines(apistructs.PipelinePageListRequest{
+		Sources:        []apistructs.PipelineSource{source},
```

Comment:
Do you need add snippet-config labels to query?

Reply:
And I think this method is too huge for this situation.
Maybe you should add a new db method.